fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -270,9 +273,20 @@ private static Value encapsulate(@Nonnull TypeRepository typeRepository, @Nonnul
                         Verify.verify(Iterables.size(objects) == 1);
                         return physicalOperator.eval(objects.iterator().next());
                     });
+        }
+        Verify.verify(arguments.size() == 2);
+        final Typed arg1 = arguments.get(1);
+        final Type res1 = arg1.getResultType();
+        if (comparisonType == Comparisons.Type.IN) {
+            SemanticException.check(res1.getTypeCode() == Type.TypeCode.ARRAY, SemanticException.ErrorCode.COMPARAND_TO_COMPARISON_IS_OF_COMPLEX_TYPE);
```

Comment:
The error code doesn't seem to fit with the error message right?

Maybe you want to check two things for this to work:
1. that `res1` is indeed an array (throw an error like type-mismatch if otherwise).
2. the `elementType` of the array is primitive (throw an error with `SemanticException.ErrorCode.COMPARAND_TO_COMPARISON_IS_OF_COMPLEX_TYPE` error code if otherwise).

Reply:
The check for the `elementType` being one of the primitive type (boolean, byteArray, double, Float, Long, Int, String, Enum) and that all the elements are of the same type happens in the `ListComparison`

Reply:
changed the error type for 1